src/uct/ib/ud/base/ud_iface.c

```diff
@@ -1045,10 +1093,20 @@ void uct_ud_iface_send_completion(uct_ud_iface_t *iface, uint16_t sn,
                                   int is_async)
 {
     uct_ud_ctl_desc_t *cdesc;
+    khiter_t it;
 
-    ucs_queue_for_each_extract(cdesc, &iface->tx.outstanding_q, queue,
-                               UCS_CIRCULAR_COMPARE16(cdesc->sn, <=, sn)) {
-        uct_ud_iface_ctl_skb_complete(iface, cdesc, is_async);
+    if (uct_ib_iface_device(&iface->super)->ordered_send_comp) {
```

Comment:
maybe it is worth to cache it somewhere on iface to avoid extra cache miss? (since it is a fast path)

Reply:
fixed